File: ffmpeg/android-ffmpeg/src/main/java/nl/bravobit/ffmpeg/ShellCommand.java
package nl.bravobit.ffmpeg;

import java.util.Arrays;
import java.util.Map;

class ShellCommand {

    Process run(String[] commandString, Map<String, String> environment) {
        Process process = null;
        try {
            ProcessBuilder processBuilder = new ProcessBuilder(commandString);
            if (environment != null) {
                processBuilder.environment().putAll(environment);
            }
            process = processBuilder.start();
        } catch (Throwable t) {
            Log.e("Exception while trying to run: " + Arrays.toString(commandString), t);
        }
        return process;
    }

}

File: ffmpeg/android-ffmpeg/src/main/java/nl/bravobit/ffmpeg/ShellCommand.kt
package nl.bravobit.ffmpeg

internal class ShellCommand {
    fun run(commandString: Array<String?>, environment: Map<String?, String?>?): Process? {
        var process: Process? = null
        try {
            val processBuilder = ProcessBuilder(*commandString)
            if (environment != null) {
                processBuilder.environment().putAll(environment)
            }
            process = processBuilder.start()
        } catch (t: Throwable) {
            Log.e("Exception while trying to run: " + commandString.contentToString(), t)
        }
        return process
    }
}